### PR TITLE
ci: Add credentials for maintainer PR notifier

### LINKED ISSUE #25742: CI/maintainer PR notifier may be lacking some creds
The pr notifier failed today with

```console

github.GithubException.RateLimitExceededException: 403 {"message": "API rate limit exceeded for 13.64.10.16. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)", "documentation_url":

```

altho it should be authenticated as it has:

```console

 10   pr_notifier:
 11     permissions:
 12       contents: read # to fetch code (actions/checkout)                                                                                                                       
 13       statuses: read # for pr_notifier.py                                                                                                                                     
 14       pull-requests: read # for pr_notifier.py   
```

not sure if something is missing here or just a github transient - rerunning worked

### PR DESCRIPTION
**Pull request by phlax**
Fix #25742 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @joycebrum

**Review by joycebrum — COMMENTED** [review]
the needs_api_review also makes a github api requisition, it should have a Authorization="Bearer $GITHUB_TOKEN" on the header too I think

**Comment by phlax** [comment]
hmm, i see - lmc if i can find a way to just use the api with the credentials loaded in the gh object ...

**Comment by phlax** [comment]
seems possible:

```console
(Pdb) pr_info.get_commits()[0].get_statuses()
<github.PaginatedList.PaginatedList object at 0x7fc476717ed0>
```

need to make sure i get the right status i think

File: .github/actions/pr_notifier/pr_notifier.py
Comment (phlax) [inline .github/actions/pr_notifier/pr_notifier.py:129]: i think this probably increases the api call count - i dont see a way to directly get status for a commit sha using the api

i think its probably still preferable to not use the private method to get this

Comment (phlax) [inline .github/actions/pr_notifier/pr_notifier.py:129]:
```suggestion
    status = repo.get_commit(pr_info.head.sha).get_statuses()
    return status[0].state == "pending" if status else False
```

Comment (phlax) [reply .github/actions/pr_notifier/pr_notifier.py]: `.totalCount` looks safest

**Review by alyssawilk — APPROVED** [review]
LGTM assuming you've tested locally

**Comment by phlax** [comment]
> LGTM assuming you've tested locally

to the extent i can - my intention is to shift this to use a runner and add unit tests - i will run it as soon as it lands to make sure nothing broke

**Pull request merged**
